app/src/main/java/com/itsaky/androidide/ExtendedMenuToolbar.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ *  This file is part of AndroidIDE.
+ *
+ *  AndroidIDE is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  AndroidIDE is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *   along with AndroidIDE.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.itsaky.androidide
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.res.Configuration
+import android.util.AttributeSet
+import android.util.Log
+import android.view.MenuItem
+import android.view.ViewTreeObserver
+import androidx.appcompat.view.menu.MenuItemImpl
+import androidx.appcompat.widget.ActionMenuView
+import androidx.core.view.children
+import androidx.core.view.iterator
+import com.google.android.material.appbar.MaterialToolbar
+import kotlin.math.floor
+
+// TODO Ask Akash which package this class should belong to and move it there
```

Comment:
Custom views are located in the `com.itsaky.androidide.ui` package. This class can be moved to the `:common` module so that it can be used by other modules as well.